lib/KettleServer.js

```diff
@@ -0,0 +1,312 @@
+/*
+Kettle Server component, corresponding directly with an express/node HTTP server
+
+Copyright 2012-2013 OCAD University
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/fluid-project/kettle/blob/master/LICENSE.txt
+*/
+
+"use strict";
+
+var fluid = require("infusion"),
+    http = require("http"),
+    https = require("https"),
+    express = require("express"),
+    kettle = fluid.registerNamespace("kettle");
+
+fluid.defaults("kettle.server", {
+    gradeNames: ["fluid.component"],
+    mergePolicy: {
+        rootMiddleware: "noexpand"
+    },
+    rootMiddleware: { // free hash of middleware namespaces to components
+        urlencoded: {
+            middleware: "{middlewareHolder}.urlencoded",
+            priority: "before:json"
+        },
+        json: {
+            middleware: "{middlewareHolder}.json"
+        }
+    },
+    invokers: {
+        stop: {
+            funcName: "kettle.server.stop",
+            args: "{that}"
+        },
+        trackConnections: {
+            funcName: "kettle.server.trackConnections",
+            args: ["{that}.sockets", "{that}.httpServer"]
+        },
+        closeConnections: {
+            funcName: "kettle.server.closeConnections",
+            args: "{that}.sockets"
+        }
+    },
+    components: {
+        middlewareHolder: {
+            type: "kettle.standardMiddleware"
+        },
+        httpRouter: {
+            type: "kettle.router.http"
+        }
+    },
+    members: {
+        expressApp: "@expand:kettle.server.makeExpressApp()",
+        httpServer: "@expand:kettle.server.httpServer({that}.expressApp)",
+        dispatcher: "@expand:kettle.server.getDispatcher({that}, {that}.rootSequence)",
+        rootSequence: "@expand:kettle.middleware.getHandlerSequence({that}, rootMiddleware, {that}.options.rootMiddleware)",
```

Comment:
We've got an extra 3rd argument here -- kettle.middleware.getHandlerSequence() [https://github.com/amb26/kettle/blob/KETTLE-32/lib/KettleMiddleware.js#L63] has 2 paramaters.